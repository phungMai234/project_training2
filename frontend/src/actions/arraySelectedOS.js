import {GET_ARRAY_SELECTED_OS} from "../constants/action-types";

const getArraySelectedOSSuccess = (data) => ({
  type: GET_ARRAY_SELECTED_OS.SUCCESS,
  payload: data
})
const getArraySelectedOSError = (message) => ({
  type: GET_ARRAY_SELECTED_OS.ERROR,
  message: message
})
export const getArraySelectedOS = (arrayOS) => {
  return dispatch => {
    if (!arrayOS) {
      dispatch(getArraySelectedOSError("Select at least OS"));
    }
    dispatch(getArraySelectedOSSuccess(arrayOS))
  }
}
