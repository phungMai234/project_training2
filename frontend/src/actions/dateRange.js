import {GET_DATE_RANGE} from "../constants/action-types";

const getDateRangeSuccess = (data) => ({
  type: GET_DATE_RANGE.SUCCESS,
  payload: data
})
const getDateRangeError = (message) => ({
  type: GET_DATE_RANGE.ERROR,
  message: message
})
export const getDateRange = (dateRange) => {
  return dispatch => {
    if (!dateRange.startDate || !dateRange.endDate) {
      dispatch(getDateRangeError("Missing date range"));
    }
    dispatch(getDateRangeSuccess(dateRange))
  }
}
