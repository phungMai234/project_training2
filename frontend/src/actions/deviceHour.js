import {GET_HOUR} from "../constants/action-types";
import axios from "axios"

const getHourRequest = () => ({
  type: GET_HOUR.PENDING
})
const getHourSuccess = (data) => ({
  type: GET_HOUR.SUCCESS,
  payload: data
})
const getHourError = (message) => ({
  type: GET_HOUR.ERROR,
  message: message
})
export const getHour = (dateRange) => {
  return dispatch => {
    dispatch(getHourRequest());
    return axios.get(`http://localhost:3001/request/device_hour?startDate=${dateRange.startDate}&endDate=${dateRange.endDate}`)
      .then(res => {
        return dispatch(getHourSuccess(res.data))
      })
      .catch(err => dispatch(getHourError(err.message)));
  }
}
