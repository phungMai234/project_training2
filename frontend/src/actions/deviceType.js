import {GET_DEVICE_TYPE} from "../constants/action-types";
<<<<<<< HEAD
import {getTypes} from "../api/deviceType-api";

export const  getDeviceType = async (dispatch) => {
    const res = await getTypes();
    return dispatch({
      type: GET_DEVICE_TYPE,
      res
    })
=======

export const getDeviceType = (types) =>{
    return {
        type: GET_DEVICE_TYPE,
        types: types
    }
>>>>>>> 3192a90e76a73f76f3377cd371d1f1101668d4e5
}
