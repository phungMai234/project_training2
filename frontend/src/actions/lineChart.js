import {GET_DATA_LINE_CHART} from "../constants/action-types";
import axios from "axios"

const getDataLineChartRequest = () => ({
  type: GET_DATA_LINE_CHART.PENDING
})
const getDataLineChartSuccess = (data) => ({
  type: GET_DATA_LINE_CHART.SUCCESS,
  payload: data
})
const getDataLineChartError = (message) => ({
  type: GET_DATA_LINE_CHART.ERROR,
  message: message
})
export const getDataLineChart = (dateRange, arrayOS, typeDate) => {
  return dispatch => {
    dispatch(getDataLineChartRequest());
    return axios.get(`http://localhost:3002/request/line_chart?startDate=${dateRange.startDate}&endDate=${dateRange.endDate}&arrayOS=${arrayOS}&date=${typeDate}`)
      .then(res => {
        dispatch(getDataLineChartSuccess(res.data))
      })
      .catch(err => {
        dispatch(getDataLineChartError(err.message))
      });
  }
}
