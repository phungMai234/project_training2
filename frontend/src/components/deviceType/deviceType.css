
.device-type {
    padding: 20px 0px;
}

.chart {
    display: flex;
    justify-content: center;
}

.box-custom {
    height: 5px;
    width: 10px;
    border-radius: 5px;
    margin-right: 10px;
}

.box-custom.android {
    background-color: #00cc99;
}

.box-custom.ios {
    background-color: #8c1aff;
}

.title-pie {
    text-align: center;
    display: flex;
    align-items: center;
    margin: 10px 0px 0px 0px;
    font-size: 12px;
}