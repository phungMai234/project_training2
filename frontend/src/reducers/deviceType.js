import {ERROR, GET_DEVICE_TYPE} from "../constants/action-types";

export const deviceTypeReducer = (state = [], action) => {
  switch (action.type) {
    case GET_DEVICE_TYPE: {
      return {...state, iOS: action.payload.iOS, android: action.payload.android}
    }
    case ERROR:{
      return state;
    }
    default:
      return state;
  }
}