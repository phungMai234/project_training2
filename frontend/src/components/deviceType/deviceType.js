import React, {useEffect} from 'react';
import './deviceType.css';
import {connect} from "react-redux"
import {VictoryPie} from "victory"
import {getTypes} from "../../actions/deviceType";
import {Spinner} from "../spinner/spinner";


function DeviceType(props) {

  let iOS = props.types.iOS;
  let android = props.types.android;
  let isFetching = props.isFetching;

  useEffect(() => {
    const fetchData = () => {
      return props.getTypes();
    }
    fetchData();

    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <div className="device-type">
      <label>Devive Type</label>
      {
        isFetching ? <Spinner/>
          : <div className="chart">
            <svg width={300} height={300}>
              <circle cx={150} cy={150} r={50} fill="#ffffff"/>
              <VictoryPie
                standalone={false}
                colorScale={["#00cc99", "#8c1aff"]}
                width={300} height={300}
                innerRadius={75}
                data={[
                  {x: " ", y: android},
                  {x: " ", y: iOS}
                ]}
              />
            </svg>
            <div>
              <div>
                <div className="title-pie">
                  <div className="box-custom ios">&nbsp;</div>
                  <label>iOS</label>
                </div>
                <label>{iOS} %</label>
              </div>
              <div>
                <div className="title-pie">
                  <div className="box-custom android">&nbsp;</div>
                  <label>Android</label>
                </div>
                <label>{android} %</label>
              </div>
            </div>
          </div>
      }

    </div>
  );

}

const mapStateToProps = (state) => ({
  types: state.types.data,
  isFetching: state.types.isFetching
})

const mapDispatchToProps = dispatch => ({
  getTypes: () => dispatch(getTypes())
})

export default connect(mapStateToProps, mapDispatchToProps)(DeviceType);