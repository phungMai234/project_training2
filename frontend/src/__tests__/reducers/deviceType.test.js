import {deviceTypeReducer} from "../../reducers/deviceType";
import {getDeviceType} from "../../actions/deviceType";

describe("decive type reducer", () => {
  const initialState = []
  it("should get device types to empty list", () => {
    expect(deviceTypeReducer(initialState, getDeviceType())).toMatchSnapshot();
  })
  it('should handle unknown actions', () => {
    expect(deviceTypeReducer(initialState, { type: 'FAKE' })).toBe(initialState);
  });
  it('should handle error', () => {
    expect(deviceTypeReducer(initialState, { type: 'ERROR', message:"error" })).toBe(initialState);
  });
})