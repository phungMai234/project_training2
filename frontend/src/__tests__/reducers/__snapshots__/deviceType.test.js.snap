// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decive type reducer should get device types to empty list 1`] = `Array []`;
